### DISCUSSION TITLE
ambiguous column name 'id'

### DISCUSSION THREAD

**Discussion by alensiljak** [SeaQL/sea-query#546] [Q&A] [answered]
Hi! Thanks a lot for this fantastic library. The most of my time on converting a rather simple project was spent on trying out, hitting walls, and rewriting the DAL code.
Hopefully, the remaining issues can be clarified or polished relatively painlessly.

I have an inner join on a select statement involving two tables, both of which have the id field (obviously). When the query is executed, the error message in the title is thrown.

In all the code examples I see that only the column names are listed, without the table name. Is there a way to require the columns to be output as `"table_name.id"` instead of just `"id"`?

I'm using rusqlite as the underlying provider.

Cheers

**Comment by alensiljak** [comment]
Actually, using double quotes won't work as that returns the quoted string as the column value.

**Comment by ikrivosheev** [comment]
@alensiljak hey! Can you show your code?)

**Reply by alensiljak** [reply]
Hi! Sure, [this](https://github.com/alensiljak/pricedb-rust/blob/main/src/database/dal_rusqlite.rs#L262) is the place where I was generating the Select statement using the query builder. Below is the workaround SQL with manual modification.

**Reply by Sytten** [reply]
Do this
```rust
fn get_security_columns() -> Vec<(SecurityIden, SecurityIden)> {
    vec![
        (SecurityIden::Table, SecurityIden::Id),
        ...
    ]
}
```

**Reply by alensiljak** [reply]
Fantastic! Please ignore my previous (deleted) comments. I'm too tired and had it written differently. Porting this project has been too much fun and I've had very little sleep in the past days. :D

That works! Fantastic tip. Worth having a prominent place in the docs for the Joins.

**Reply by billy1624** [reply]
Thanks! @Sytten

**Comment by alensiljak** [comment, marked as answer]
A [working solution](https://github.com/SeaQL/sea-query/discussions/546#discussioncomment-4347348)
